docs/guides/web-services.rst

```diff
@@ -113,26 +118,63 @@ You can use additional fields to describe your parameter, e.g. ``description``.
 
 .. code:: php
 
-    elgg_ws_expose_function('test.greet',
-                    'my_greeting',
-                    array(
-                        'name' => array(
-                            'type' => 'string',
-                            'required' => true,
-                            'description' => 'Name of the person to be greeted by the API',
-                        ),
-                        'greeting' => array(
-                            'type' => 'string',
-                            'required' => false,
-                            'default' => 'Hello',
-                            'description' => 'Greeting to be used, e.g. "Good day" or "Hi"',
-                        ),
-                    ),
-                    'A testing method which greets the user with a custom greeting',
-                    'GET',
-                    false,
-                    false
-    );
+	elgg_ws_expose_function(
+		'test.greet',
+		'my_greeting',
+		[
+			'name' => [
+				'type' => 'string',
+				'required' => true,
+				'description' => 'Name of the person to be greeted by the API',
+			],
+			'greeting' => [
+				'type' => 'string',
+				'required' => false,
+				'default' => 'Hello',
+				'description' => 'Greeting to be used, e.g. "Good day" or "Hi"',
+			],
+		],
+		'A testing method which greets the user with a custom greeting',
+		'GET',
+		false,
+		false
+	);
+
+Parameters as associative array
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you have a large number of method parameters, you can force the execution script
+to invoke the callback function with a single argument that contains an associative
+array of parameter => input pairs (instead of each parameter being a separate argument).
+To do that, set ``$assoc`` to ``true`` in ``elgg_ws_expose_function()``.
+
+.. code:: php
+
+	function greet_me($values) {
+		$name = elgg_extract('name', $values);
+		$greeting = elgg_extract('greeting', $values, 'Hello');
+		return "$greeting, $name";
+	}
+
+	elgg_ws_expose_function(
+		"test.greet",
+		"greet_me",
+		[
+			"name" => [
+				'type' => 'string',
+			],
+			"greeting" => [
+				'type' => 'string',
+				'default' => 'Hello',
+				'required' => false,
+			],
+		],
+		'A testing method which echos a greeting',
+		'GET',
+		false,
+		false,
+		true // $assoc makes the callback receive an associative array
```

Comment:
Individual parameters can be arrays, too, so we can't tell by reflection. We could better do this by having plugins just provide an instance of an interface (with isAssoc(), etc.) for each method, but the goal here is to nudge this along instead of getting bogged down.